starlette/responses.py

```diff
@@ -419,10 +413,14 @@ async def _handle_multiple_ranges(
         else:
             async with await anyio.open_file(self.path, mode="rb") as file:
                 for start, end in ranges:
-                    await file.seek(start)
-                    chunk = await file.read(min(self.chunk_size, end - start))
                     await send({"type": "http.response.body", "body": header_generator(start, end), "more_body": True})
-                    await send({"type": "http.response.body", "body": chunk, "more_body": True})
+                    while True:
+                        await file.seek(start)
+                        chunk = await file.read(min(self.chunk_size, end - start))
+                        start += len(chunk)
+                        await send({"type": "http.response.body", "body": chunk, "more_body": True})
+                        if len(chunk) < self.chunk_size or start >= end:
```

Comment:
Is the `len(chunk) < self.chunk_size` necessary? I'm too tired, I'll check tomorrow. Just leaving my own comment here.

Reply:
it can be mathematically deduced that is not required

Reply:
Yes, there is no need to check once `len(chunk) < self.chunk_size`